api/workload/x509_client_test.go

```diff
@@ -2,218 +2,291 @@ package workload
 
 import (
 	"context"
-	"crypto/x509"
 	"errors"
-	"io/ioutil"
 	"net"
-	"os"
-	"path"
-	"reflect"
-	"sync"
 	"testing"
 	"time"
 
+	"github.com/andres-erbsen/clock"
+	"github.com/gogo/protobuf/proto"
 	"github.com/spiffe/spire/proto/api/workload"
-	"github.com/spiffe/spire/test/util"
+	"github.com/spiffe/spire/test/fakes/fakeworkloadapi"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
 
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/metadata"
+const (
+	// No tests rely on timing, however, due to the nature of the X509Client
+	// and StreamX509SVID calls the only way to observe externally that
+	// something is broken is through failure to receive on channels. Using the
+	// following timeout to prevent having to wait for the default go test
+	// timeout (10 minutes) if this happens. The timeout should be large enough
+	// by a comfortable margin to accomodate for slower running platforms, like
+	// Travis CI.
+	testTimeout = time.Minute
 )
 
-func TestClient_StartAndStop(t *testing.T) {
-	sockPath, grpcServer, handler := newStubbedAPI(t)
-	defer grpcServer.Stop()
-	defer os.RemoveAll(path.Dir(sockPath))
+var (
+	responseA = &workload.X509SVIDResponse{Svids: []*workload.X509SVID{{SpiffeId: "A"}}}
+	responseB = &workload.X509SVIDResponse{Svids: []*workload.X509SVID{{SpiffeId: "B"}}}
+	responseC = &workload.X509SVIDResponse{Svids: []*workload.X509SVID{{SpiffeId: "C"}}}
 
-	addr := &net.UnixAddr{
-		Net:  "unix",
-		Name: sockPath,
-	}
-	config := &X509ClientConfig{
-		Addr:    addr,
-		Timeout: 5 * time.Second,
-	}
-	c := NewX509Client(config)
+	fakeError = errors.New("fake error")
```

Comment:
nit: golint would complain about the naming here. By convention, error variables start with "err"
https://github.com/golang/go/wiki/Errors#naming

Reply:
Sounds good to me. I'll update the name.